test/e2e/util/util.go

```diff
@@ -983,3 +986,35 @@ func RunQueryAtTime(ctx context.Context, log logr.Logger, prometheusClient prome
 		},
 	}, nil
 }
+
+func EnsurePodsWithEmptyDirPVsHaveSafeToEvictAnnotations(t *testing.T, ctx context.Context, hostClient crclient.Client, hcpNs string) {
+	t.Run("EnsurePodsWithEmptyDirPVsHaveSafeToEvictAnnotations", func(t *testing.T) {
+		g := NewWithT(t)
+		hcpPods := &corev1.PodList{}
+		if err := hostClient.List(ctx, hcpPods, &client.ListOptions{
+			Namespace: hcpNs,
+		}); err != nil {
+			t.Fatalf("cannot list hostedControlPlane pods: %v", err)
+		}
+
+		// Check if the pod's volumes are hostPath or emptyDir, if so, the deployment
+		// should annotate the pods with the safe-to-evict-local-volumes, with all the volumes
+		// involved as an string  and separated by comma, to satisfy CA operator contract.
+		// more info here: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-types-of-pods-can-prevent-ca-from-removing-a-node
+		for _, pod := range hcpPods.Items {
+			localVolumeList := make([]string, 0)
+			for _, volume := range pod.Spec.Volumes {
+				// Check the volumes, if they are emptyDir or hostPath, should be annotated
+				if volume.EmptyDir != nil || volume.HostPath != nil {
+					localVolumeList = append(localVolumeList, volume.Name)
+				}
+			}
+			// After going through all the volumes in the pod, validates if
+			// the annotation value makes sense with the expectations.
+			podSafeToEvictVolumes := fmt.Sprintf("\"%s\"", strings.Join(localVolumeList, ","))
+			if _, exists := pod.ObjectMeta.Annotations[suppconfig.PodSafeToEvictLocalVolumesKey]; exists {
+				g.Expect(pod.ObjectMeta.Annotations[suppconfig.PodSafeToEvictLocalVolumesKey]).To(Equal(podSafeToEvictVolumes), "pod with name %s do not have the right volumes set in the safe-to-evict-local-volume annotation: \nCurrent: %s, Expected: %s", pod.Name, pod.ObjectMeta.Annotations[suppconfig.PodSafeToEvictLocalVolumesKey], podSafeToEvictVolumes)
+			}
+		}
```

Comment:
this breaks KubeVirt platform testing due to the virt-launcher pods being in the HCP namespace

```
=== RUN   TestCreateCluster/EnsurePodsWithEmptyDirPVsHaveSafeToEvictAnnotations
    util.go:1019: 
        the pod virt-launcher-example-rbz87-lkd97-j8zrm has not the safe-to-evict annotations properly set: map[k8s.ovn.org/pod-networks:{"default":{"ip_addresses":["10.131.0.168/23"],"mac_address":"0a:58:0a:83:00:a8","gateway_ips":["10.131.0.1"],"ip_address":"10.131.0.168/23","gateway_ip":"10.131.0.1"}} k8s.v1.cni.cncf.io/network-status:[{
            "name": "ovn-kubernetes",
            "interface": "eth0",
            "ips": [
                "10.131.0.168"
            ],
            "mac": "0a:58:0a:83:00:a8",
            "default": true,
            "dns": {}
        }] kubectl.kubernetes.io/default-container:compute kubevirt.io/domain:example-rbz87-lkd97 kubevirt.io/migrationTransportUnix:true openshift.io/scc:privileged post.hook.backup.velero.io/command:["/usr/bin/virt-freezer", "--unfreeze", "--name", "example-rbz87-lkd97", "--namespace", "e2e-clusters-h69pd-example-rbz87"] post.hook.backup.velero.io/container:compute pre.hook.backup.velero.io/command:["/usr/bin/virt-freezer", "--freeze", "--name", "example-rbz87-lkd97", "--namespace", "e2e-clusters-h69pd-example-rbz87"] pre.hook.backup.velero.io/container:compute seccomp.security.alpha.kubernetes.io/pod:localhost/kubevirt/kubevirt.json]
        Expected
            <bool>: false
        to be true
```

Reply:
Ok, let me rethink about the e2e test, maybe following the @enxebre advice about the pod list to have the annotation

Reply:
@jparrill if this broke with your previous test it means the code is not working as expected, that pod wasn't meeting the expectation

Reply:
The think is I didn't check with kubevirt provider, this is why I didn't realize. I think we need to add the annotation also to that provider deployments. WDYT? @enxebre @davidvossel

Reply:
if you don't annotate this pod then the purpose of the ticket would be defeated, that pod would block (auto)scale down of management cluster. David, what's the volume purpuse in that pod, any concern adding the annotation?

Reply:
> David, what's the volume purpuse in that pod, any concern adding the annotation?

It's the pod that runs the qemu VMs. The empty dir is used to host some unix sockets that are shared across containers in the pod.

If nodepools are being scaled down to 0, then these pods will naturally disappear.

Even outside of that, it's unlikely we need to worry about pod eviction of virt-launcher pods. There's an entire system in place to handle eviction for those pods within KubeVirt. those virt-launcher pods are hosting qemu virtual machines. We've got logic that live migrates the VMs to another pod during eviction and then gracefully shuts down the pod

Reply:
@davidvossel I've added in the PR the Kubevirt part, just if you wanna take a quick look.